packages/shared/src/components/fields/MarkdownInput/index.tsx

```diff
@@ -172,6 +172,119 @@ function MarkdownInput(
     focusInput(textareaRef.current, [content.length, content.length]);
   }, []);
 
+  const handleMarkdownShortcut = (
+    e: React.KeyboardEvent<HTMLTextAreaElement>,
+  ) => {
+    const textarea = textareaRef.current;
+    const { selectionStart, selectionEnd, value } = textarea;
+
+    const updateTextarea = (
+      newValue: string,
+      newStart: number,
+      newEnd: number,
+    ) => {
+      onValueUpdate?.(newValue);
+      callbacks.onInput?.({
+        currentTarget: { value: newValue },
+      } as React.FormEvent<HTMLTextAreaElement>);
+
+      requestAnimationFrame(() => {
+        textarea.value = newValue;
+        textarea.setSelectionRange(newStart, newEnd);
+      });
+    };
+
+    const handleTextFormatting = (symbol: string) => {
+      const selectedValue = value.substring(selectionStart, selectionEnd);
+      const trimmedValue = selectedValue.trim();
+
+      const leadingWhitespace =
+        selectedValue.length - selectedValue.trimStart().length;
+      const newStart = selectionStart + leadingWhitespace;
+      const newEnd = newStart + trimmedValue.length;
+      const symbolLength = symbol.length;
+
+      const isFormatted =
+        value.substring(newStart - symbolLength, newStart) === symbol &&
+        value.substring(newEnd, newEnd + symbolLength) === symbol;
+
+      let updatedValue;
+      let updatedStart;
+      let updatedEnd;
+
+      if (isFormatted) {
+        updatedValue =
+          value.substring(0, newStart - symbolLength) +
+          trimmedValue +
+          value.substring(newEnd + symbolLength);
+        updatedStart = newStart - symbolLength;
+        updatedEnd = newEnd - symbolLength;
+      } else {
+        updatedValue =
+          value.substring(0, newStart) +
+          symbol +
+          trimmedValue +
+          symbol +
+          value.substring(newEnd);
+        updatedStart = newStart + symbolLength;
+        updatedEnd = newEnd + symbolLength;
+      }
+
+      updateTextarea(updatedValue, updatedStart, updatedEnd);
+    };
+
+    const handleLinkPaste = () => {
+      navigator.clipboard.readText().then((clipboardText) => {
+        const selectedText = value.substring(selectionStart, selectionEnd);
+        const trimmedText = selectedText.trim();
+
+        const leadingWhitespace =
+          selectedText.length - selectedText.trimStart().length;
+        const trailingWhitespace =
+          selectedText.length - selectedText.trimEnd().length;
+
+        const newStart = selectionStart + leadingWhitespace;
+        const newEnd = selectionEnd - trailingWhitespace;
+
+        const newText =
+          /^https?:\/\//.test(clipboardText) && trimmedText
```

Comment:
I think we have an existing function that checks for valid http url. Just so to keep the validations consistent.
```suggestion
          isValidHttpUrl(clipboardText) && trimmedText
```